whylog/config/rule.py

```diff
@@ -4,13 +4,24 @@
 import six
 
 from whylog.config.parsers import RegexParserFactory
+from whylog.constraints.verifier import Verifier
 
 
 class Rule(object):
     def __init__(self, causes, effect, constraints):
         self._causes = causes
         self._effect = effect
         self._constraints = constraints
+        self._frequency_information = self._gather_causes_frequency_information()
+
+    def _gather_causes_frequency_information(self):
+        causes_with_frequency_info = [(self._causes[0].name, 0)]
+        for cause in self._causes:
```

Comment:
I agree that we can sorted causes list, but we must remeber to update constraints properly